### PR TITLE
fix(ci): Do not skip step for visual diffs on PRs or master

### PR DESCRIPTION
**Pull request by armenzg**
This fixes a regression from https://github.com/getsentry/sentry/pull/33609 since `head_repository.full_name` is not part of the `github` context in PRs.

Visual Snapshots have not been running from acceptance/visual-diff for the last week. In the PR,
we missed that it did not get triggered:
https://github.com/getsentry/sentry/actions/runs/2167609568

The step (rather than the job) is being skipped on PRs:
https://github.com/getsentry/sentry/runs/6117835780?check_suite_focus=true

This has not been noticed because `visual-diff.yml` would still trigger Visual Snapshots for PRs:
https://github.com/getsentry/sentry/pull/33835/checks?check_run_id=6117841569

### CODE REVIEW

**Comment by armenzg** [comment]
I see now the ["Diff Snapshots" step](https://github.com/getsentry/sentry/runs/6163597815?check_suite_focus=true) triggering.

I also see the [visual-diff job](https://github.com/getsentry/sentry/actions/runs/2222324603) triggering because this has not yet merged to master.

**Review by asottile-sentry — APPROVED** [review]
good catch!

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -324,8 +324,8 @@ jobs:
       - name: Diff snapshots
         id: visual-snapshots-diff
         uses: getsentry/action-visual-snapshot@v2
-        # Do not execute on forks or on master. Forks are handled in visual-diff.yml
-        if: github.head_repository.full_name == 'getsentry/sentry' && github.ref != 'refs/heads/master'
+        # Do not execute on forks. Forks are handled in visual-diff.yml
+        if: github.event.pull_request.head.repo.full_name == 'getsentry/sentry' || github.ref == 'refs/heads/master'
```
Comment (armenzg) [inline .github/workflows/acceptance.yml:328]: I've pushed one more commit.
This needs to run on PRs within Sentry or a master push.
`visual-diff` is exclusive for PRs from forks.

This [push](https://github.com/getsentry/github-actions-testing/actions/runs/2227071984) on `master` shows this running and [workflow_run not executing](https://github.com/getsentry/github-actions-testing/actions/runs/2227075158):
image
<br />
image

**Review by asottile-sentry — APPROVED** [review]

**Comment by armenzg** [comment]
Now that this is merged, PRs will skip the workflow_run of `visual-diff`:
image

There's a risk of introducing an acceptance regressions while devs update their branches. I've notified on Slack. I'm willing to take this risk since VS are already flaky and not a required check.

**Comment by armenzg** [comment]
I see [this PR](https://github.com/getsentry/sentry/runs/6178356300?check_suite_focus=true) triggering Visual Snapshots:
image

image

**Pull request merged**